library/common/jni/jni_interface.cc

```diff
@@ -60,11 +83,16 @@ static void jvm_on_exit(void*) {
 }
 
 extern "C" JNIEXPORT jint JNICALL Java_io_envoyproxy_envoymobile_engine_JniLibrary_runEngine(
-    JNIEnv* env, jclass, jlong engine, jstring config, jstring jvm_log_level, jobject context) {
-  jobject retained_context = env->NewGlobalRef(context); // Required to keep context in memory
-  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit, retained_context};
+    JNIEnv* env, jclass, jlong engine, jstring config, jstring jvm_log_level, jobject context,
+    jobject envoy_logger_context) {
+  jobject retained_context_on_start =
+      env->NewGlobalRef(context); // Required to keep context in memory
+  jobject retained_context_on_log =
+      env->NewGlobalRef(envoy_logger_context); // Required to keep context in memory
+  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit,
+                                             retained_context_on_start};
   // TODO(junr03): wire up once Android support lands.
-  envoy_logger logger = {nullptr, nullptr, nullptr};
+  envoy_logger logger = {jvm_on_log, jvm_on_log_exit, retained_context_on_log};
```

Comment:
We should only be building this if the user has `setLogger` in the engine builder